services/priceService.ts

```diff
@@ -62,23 +61,121 @@ export async function upsertCurrentPricesForNetworkId (responseData: IResponseDa
   }
-  const todayString = moment().format(PRICE_API_DATE_FORMAT)
+  if (appInfo.priceAPIToken === '') {
+    throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_TOKEN_400.message)
+  }
+}
+
+function getPriceByDayURL (day: moment.Moment, ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
+  return `${appInfo.priceAPIURL}/pricebydate/${appInfo.priceAPIToken}/${ticker}+${day.format(PRICE_API_DATE_FORMAT)}`
+}
+
+function getAllPricesByTickerURL (ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
+  return `${appInfo.priceAPIURL}/dailyprices/${appInfo.priceAPIToken}/${ticker}`
+}
+
+export async function getPriceForDayTicker (day: moment.Moment, ticker: string, attempt: number = 1): Promise<IResponseData | null> {
+  try {
+    const res = await axios.get(getPriceByDayURL(day, ticker), {
+      timeout: PRICE_API_TIMEOUT
+    })
+
+    if (res.data.success !== false) { return res.data } else { return null }
+  } catch (error) {
+    console.error(`Problem getting price of ${ticker} ${day.format('YYYYMMDD')} -> ${error as string} (attempt ${attempt})`)
+
+    if (attempt < PRICE_API_MAX_RETRIES) {
+      return await getPriceForDayTicker(day, ticker, attempt + 1)
+    } else { return null }
+  }
+}
+
+export async function getAllPricesByTicker (ticker: string, attempt: number = 1): Promise<IResponseDataDaily[] | null> {
+  try {
+    const res = await axios.get(getAllPricesByTickerURL(ticker), {
+      timeout: PRICE_API_TIMEOUT
+    })
+
+    if (res.data.success !== false) {
+      const dailyPrices: IResponseDataDaily[] = Object.entries<IResponseData>(res.data).map(e => {
+        return {
+          day: e[0],
+          ...e[1]
+        }
+      })
+      return dailyPrices
+    } else { return null }
+  } catch (error) {
+    console.error(`Problem getting price of ${ticker} -> ${error as string} (attempt ${attempt})`)
+
+    if (attempt < PRICE_API_MAX_RETRIES) {
+      return await getAllPricesByTicker(ticker, attempt + 1)
+    } else { return null }
+  }
+}
+
+export async function syncPastPrices (): Promise<void> {
+  const lastTimestamp = await prisma.price.findFirst({
+    where: {
+      networkId: XEC_NETWORK_ID, // any valid network
+      quoteId: USD_QUOTE_ID // any valid quote
+    },
+    orderBy: [{ timestamp: 'desc' }],
+    select: { timestamp: true }
+  })
+
+  if (lastTimestamp?.timestamp === undefined) throw new Error('No prices found in the worker, please run prisma seed')
+  const lastDateInDB = moment.unix(lastTimestamp?.timestamp)
+
+  const date = moment().startOf('day')
+  const datesToRetrieve: moment.Moment[] = []
+
+  while (date.isAfter(lastDateInDB)) {
+    datesToRetrieve.push(date.clone())
+    date.add(-1, 'day')
+  }
+
+  await Promise.all(
+    Object.values(TICKERS).map(async (ticker) =>
+      datesToRetrieve.map(async date => {
+        const price = await getPriceForDayTicker(date, ticker)
+        if (price != null) {
```

Comment:
The `getPriceForDayTicker` already tries many times to avoid network error, but if the api gives us a null price because it is broken for that day or doesn't have its data, then throwing an error here would kill the worker and avoid syncing the other days correct prices that come back from the api, that `if` allows that. 

Since we can't automatically do nothing if the api doesn't give us the price, better have 1 day hole than 5 day hole.

That being said, I will include a `console.error` message here to warn about this.